specs/Print.md

```diff
@@ -0,0 +1,816 @@
+
+  /// Provides the Pdf data of current web page asynchronously for the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings. Passing
+  /// nullptr for `printSettings` results in default print settings used.
+  ///
+  /// The async `PrintToPdfStream` operation completes when it finishes
+  /// writing to the stream. At this time the `ICoreWebView2PrintToPdfStreamCompletedHandler`
+  /// is invoked. Only one `Printing` operation can be in progress at a time. If
+  /// `PrintToPdfStream` is called while a `PrintToPdfStream` or `PrintToPdf` or `Print` or `ShowPrintUI`
+  /// job is in progress, the completed handler is immediately invoked with `E_ABORT`.
+  ///
+  /// \snippet AppWindow.cpp PrintToPdfStream
+  HRESULT PrintToPdfStream([in] ICoreWebView2PrintSettings2* printSettings,
+                           [in] ICoreWebView2PrintToPdfStreamCompletedHandler* handler);
+}
+
+/// Receives the result of the `Print` method.                               |
+[uuid(4B2FE043-1A1A-4924-803D-C05E5FDCA3B0), object, pointer_default(unique)]
+interface ICoreWebView2PrintCompletedHandler : IUnknown {
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] BOOL isSuccessful);
+}
+
+/// Receives the result of the `PrintToPdfStream` method.
+/// `errorCode` returns S_OK if the PrintToPdfStream operation succeeded.
+/// The printable pdf data is returned in the `pdfStream` object.
+[uuid(78A9FC37-249B-4E7E-A2E6-20CAAA14AE7B), object, pointer_default(unique)]
+interface ICoreWebView2PrintToPdfStreamCompletedHandler : IUnknown {
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] IStream* pdfStream);
+}
+
+/// Settings used by the `Print` method.
+[uuid(DA3AD25C-D121-44DC-83DA-BF5E50DBF0DF), object, pointer_default(unique)]
+interface ICoreWebView2PrintSettings2 : ICoreWebView2PrintSettings {
+  /// Page range to print. Defaults to empty string, which means print all pages.
+  ///
+  /// The PageRanges property is a list of page ranges specifying one or more pages that
+  /// should be printed separated by commas. Any whitespace between page ranges is ignored.
+  /// A valid page range is either a single integer identifying the page to print, or a range
+  /// in the form `[start page]-[last page]` where `start page` and `last page` are integers
+  /// identifying the first and last inclusive pages respectively to print.
+  /// Every page identifier is an integer greater than 0 unless wildcards are used (see below examples).
+  /// The first page is 1.
+  ///
+  /// In a page range of the form `[start page]-[last page]` the start page number must be
+  /// larger than 0 and less than or equal to the document's total page count.
+  /// If the `start page` is not present, then 1 is used as the `start page`.
+  /// The `last page` must be larger than the `start page`.
```

Comment:
So "2" is supported but "2-2" isn't. That covers all the scenarios, just a little odd, maybe means the app has to write some special-case code.

Reply:
Chromium code is handling the case 2-2 by removing the duplicated page and printing it only one time. 
Here 2-2 is also valid and prints page 2.

I'll update the last example accordingly.

Reply:
Please fix.